acc_test/test_utils.go

```diff
@@ -21,41 +21,55 @@ func getOptionalEnv(key, fallback string) string {
 }
 
 func checkClusterRunning(resourceName, hostname, username, apiKey string) resource.TestCheckFunc {
+	// wait another minute after cluster goes to RUNNING to make sure all operations will work
+	// sometimes says cluster is not ready for resizing
 	return func(s *terraform.State) error {
 		resourceState := s.Modules[0].Resources["instaclustr_cluster."+resourceName]
 		id := resourceState.Primary.Attributes["cluster_id"]
-		client := new(instaclustr.APIClient)
-		client.InitClient(hostname, username, apiKey)
+		return waitClusterStatus(id, hostname, username, apiKey, "RUNNING", 60)
+	}
+}
+
+func checkClusterRunningWithDelay(resourceName, hostname, username, apiKey string, delayAfter time.Duration) resource.TestCheckFunc {
+	// wait another minute after cluster goes to RUNNING to make sure all operations will work
+	// sometimes says cluster is not ready for resizing
+	return func(s *terraform.State) error {
+		resourceState := s.Modules[0].Resources["instaclustr_cluster."+resourceName]
+		id := resourceState.Primary.Attributes["cluster_id"]
+		return waitClusterStatus(id, hostname, username, apiKey, "RUNNING", delayAfter)
+	}
+}
 
-		const ClusterReadInterval = 5
-		const WaitForClusterTimeout = 40 * 60
-		var latestStatus string
-		timePassed := 0
-		fmt.Print("\033[s")
-		for {
-			cluster, err := client.ReadCluster(id)
-			if err != nil {
-				fmt.Printf("\n")
-				return fmt.Errorf("[Error] Error retrieving cluster info: %s", err)
-			}
-			latestStatus = cluster.ClusterStatus
-			if cluster.ClusterStatus == "RUNNING" {
-				break
-			}
-			if timePassed > WaitForClusterTimeout {
-				fmt.Printf("\n")
-				return fmt.Errorf("[Error] Timed out waiting for cluster to have the status 'RUNNING'. Current cluster status is '%s'", latestStatus)
-			}
-			timePassed += ClusterReadInterval
-			fmt.Printf("\033[u\033[K%ds has elapsed while waiting for the cluster to reach RUNNING.\n", timePassed)
-			time.Sleep(ClusterReadInterval * time.Second)
+func waitClusterStatus(clusterId, hostname, username, apiKey, expectedStatus string, delayAfter time.Duration) error {
```

Comment:
What is all this for? Isn't the capacity to wait already built into the provider?

Reply:
Historical reason for this one is this one predates the provider waiting stuff. The reason I used it now seems to be based on wrong assumption, I thought if I create 2 KC clusters at the same time and wait for running, it will create the cluster sequentially.